### ISSUE TITLE
Array{T,N}(n...) should work for any N

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#18459]
Right now, you can do `Array{T,1}(n)` and `Array{T,2}(n1, n2)` but not `Array{T,3}(n1, n2, n3)` (although `Array{T}(n1, n2, n3)` and `Array{T,3}((n1, n2, n3))` both work). This is inconsistent and potentially surprising. It should be possible to make all constructors of this form work using bound varargs.

**Comment by yuyichao** [comment]
Ref https://github.com/JuliaLang/julia/issues/14201, which was AFAIK the last time this came up.

**Comment by pkofod** [comment]
Currently building to test locally first (from fresh clone, so...), but is it something like
```julia
(::Type{Array{T,N}}){T,N}(d::Vararg{Int, N}) = ccall(:jl_new_array, Array{T,N}, (Any,Any), Array{T,N}, d)
```
right?

**Comment by SanketDG** [comment]
Was this fixed in #18745?

**Comment by yuyichao** [comment]
Yes.
